### PR TITLE
Changed out all the C++ MPI bindings for C ones

### PR DESCRIPTION
**Pull request by joshuawall**
I think I got all of them!

### CODE REVIEW

**Comment by rieder** [comment]
This one is a bit difficult to test, since I don't have seem to have access to a machine with MPI that actually removed the C++ bindings...

**Comment by joshuawall** [comment]
Steven,

     I talked with Steve about it yesterday, I think he's going to take a
look at it.

Joshua Wall

Doctoral Candidate
Department of Physics
Drexel University

**Comment by ipelupessy** [comment]
any updates on this one?

**Comment by rieder** [comment]
@stevemcmillan?

**Comment by joshuawall** [comment]
I'm not sure if Steve's looked at it, but as for testing I'd just suggest
grabbing the latest OpenMPI and do a locale install. If you install it and
just leave off --with-cxx-extensions (or something similar... trying to
recall off the top of my head) it won't put them in. In fact, I think the
default behavior of OpenMPI since v 3 is to not install them by default.
OMPI is pretty quick to install. I'd look at it myself, but I'm in the
middle of preparing my thesis defense....

Oh, did I mention I'm defending my thesis on March 15 !? :P

Joshua Wall

Doctoral Candidate
Department of Physics
Drexel University

**Comment by ipelupessy** [comment]
congratulations!!

**Comment by rieder** [comment]
Coming back to this issue, I think it's not fully solved yet. Can somebody test if ph4 works without C++ bindings, and if no such bindings remain in the code?

**Comment by ipelupessy** [comment]
this seems to work: with MPI which fails to compile the master ph4, this patch compiles& passes the ph4 tests (test_ph4.py)..not sure there is anything more we should require..

**Comment by ipelupessy** [comment]
the PR looks big, but that is because I merged master into this...

**Comment by rieder** [comment]
Thanks Inti,
I will try to have a look on Monday - first need to recover from EWASS...

File: src/amuse/community/ph4/src/gpu.cc [hunk 1]
```diff
@@ -807,7 +807,7 @@ void jdata::get_densities_on_gpu()	// under development
 #ifndef NOMPI
 	    cout << mpi_rank << " sending neighbors to" << mpi_rank-offset
 		 << endl << flush;
-	    mpi_comm.Send(pneighbors, nj*knn, MPI_INT,
+	    MPI_Send(mpi_comm,pneighbors, nj*knn, MPI_INT,
```
Comment (ipelupessy) [inline src/amuse/community/ph4/src/gpu.cc]: this is wrong, comm must come at the end

File: src/amuse/community/ph4/src/gpu.cc [hunk 2]
```diff
@@ -826,7 +826,7 @@ void jdata::get_densities_on_gpu()	// under development
         cout << mpi_rank << " receiving neighbors from"
 		     << mpi_rank+offset
 		     << endl << flush;
-		mpi_comm.Recv(pneighbors+nj*knn, nj*knn, MPI_INT,
+		MPI_Recv(mpi_comm, pneighbors+nj*knn, nj*knn, MPI_INT,
```
Comment (ipelupessy) [inline src/amuse/community/ph4/src/gpu.cc]: wrong argument order, status missing

**Comment by ipelupessy** [comment]
so some fixes are necessary - this is in code that is currently dead anyway, and its quite obvious what the error is...

**Comment by ipelupessy** [comment]
put out a rebased PR #447
(not sure if necessary, but looked better)

**Pull request closed without merging**